src/Symfony/Component/Validator/Context/ExecutionContext.php

```diff
@@ -139,10 +139,10 @@ public function setConstraint(Constraint $constraint): void
         $this->constraint = $constraint;
     }
 
-    public function addViolation(string $message, array $parameters = []): void
+    public function addViolation(string $message, array $parameters = [], string|false|null $translationDomain = null): void
     {
         $this->violations->add(new ConstraintViolation(
-            $this->translator->trans($message, $parameters, $this->translationDomain),
+            false === $translationDomain ? $message : $this->translator->trans($message, $parameters, $translationDomain ?? $this->translationDomain),
```

Comment:
this does not work properly if the default translation domain (`$this->translationDomain`) is set to `false`.

However, I'm not convinced that disabling translations globally is a feature we should support.